### ISSUE TITLE
Is this project still active/valid?

### ISSUE THREAD

**Issue by icelava** [GMMan/SteamCloudFileManagerLite#5]
I recently grew the curiosity of figuring out what exactly are being stored (raw files/folders) in our Steam Cloud storage system, and came across this project.

On attempting to compile this solution, I hit the Steamworks.NET missing dependency error which lead me to clone the repository https://github.com/rlabrecque/Steamworks.NET.git

After compiling that solution, logically with the Windows (x64) build config, I redirected SteamCloudFileManagerLite's dependency to that new assembly file. But face the error,

```
1>C:\projects\GitHub\SteamCloudFileManagerLite\SteamCloudFileManager\RemoteStorage.cs(86,20,86,83): error CS1502: The best overloaded method match for 'Steamworks.SteamRemoteStorage.GetQuota(out ulong, out ulong)' has some invalid arguments
1>C:\projects\GitHub\SteamCloudFileManagerLite\SteamCloudFileManager\RemoteStorage.cs(86,52,86,62): error CS1503: Argument 1: cannot convert from 'out int' to 'out ulong'
1>C:\projects\GitHub\SteamCloudFileManagerLite\SteamCloudFileManager\RemoteStorage.cs(86,68,86,82): error CS1503: Argument 2: cannot convert from 'out int' to 'out ulong'
```

So it appears SteamCloudFileManagerLite is meant to be AnyCPU; am I supposed to be forcing Steamworks.NET as AnyCPU as well? Its other build option is OSX-Linux which wouldn't make sense here since I'm running WIndows all the day.

**Comment by GMMan** [comment]
Master branch is pretty old, but the other branch is recent. See https://github.com/GMMan/SteamCloudFileManagerLite/commit/b0b44dc57800428f45b3333b45515ba9b5176e25 for necessary changes.

**Comment by icelava** [comment]
Ok the upload-support branch compiles. However when pressing [Connect] button it complains of a lack of CSteamworks.dll.

> ---------------------------
> Failed to connect
> ---------------------------
> System.DllNotFoundException: Unable to load DLL 'CSteamworks': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
> 
>    at Steamworks.NativeMethods.SteamAPI_Init()
> 
>    at SteamCloudFileManager.RemoteStorage..ctor(UInt32 appID) in c:\projects\GitHub\SteamCloudFileManagerLite\SteamCloudFileManager\RemoteStorage.cs:line 43
> 
>    at SteamCloudFileManager.RemoteStorage.CreateInstance(UInt32 appID) in c:\projects\GitHub\SteamCloudFileManagerLite\SteamCloudFileManager\RemoteStorage.cs:line 113
> 
>    at SteamCloudFileManager.MainForm.connectButton_Click(Object sender, EventArgs e) in c:\projects\GitHub\SteamCloudFileManagerLite\SteamCloudFileManager\MainForm.cs:line 39
> ---------------------------
> OK   
> ---------------------------

Sorry I am not a game developer so not familiar with Steamworks API so I guess have to download the Steamworks SDK 1.41 (which seems to be the only version available at https://partner.steamgames.com/home)

Thing is that SDK only has a redist DLL steam_api64.dll? I can't see any CSteamworks.dll that the Steamworks.NET wrapper seemingly wants to DllImport....

**Comment by GMMan** [comment]
Binary dependencies can be found [here](https://github.com/rlabrecque/Steamworks.NET/tree/master/Plugins/x86_64).

If you'd rather use a prebuilt copy, it is available [here](https://github.com/GMMan/SteamCloudFileManagerLite/releases/tag/v1.0.0.0-alpha-3).

**Comment by icelava** [comment]
Oh so the Steamworks.NET project build definition did not include those raw library files in \Plugins\x86_64 to be copied over to the final output folder. Odd.

Anyway it seems to run without error now, but I see I needed to launch this on my dedicated gaming computer (where Steam is installed, and able to borrow my Steam identity). My laptop is only for mobile work/development where I run Visual Studio.

However I am not entirely surely if everything is working properly, as I input a game's appID like _20900_ - **The Witcher Enhanced Edition** - it can report the consumed storage vs quota limit, but does not list any file item at all to inform what files are taking up that space....

Am I missing out on something with this behaviour/result?

**Comment by GMMan** [comment]
Some games use autocloud, and those files are not exposed on the Steam Cloud interface. You can check by going to [SteamDB](https://steamdb.info/), going to an app page, and clicking on the "User Cloud" tab. If there is a "savefiles" section, the game is using autocloud and will likely not show any files through the Steam Cloud interface.

**Comment by icelava** [comment]
Sorry what does autocloud really mean? What salient feature set does that indicate?

**The Witcher Enhanced Edition** does sync save files to cloud storage.
https://steamdb.info/app/20900/ufs/

However the problem is this game - because of its highly fragile quest dependency design - forces players to save the game very frequently. A single chapter of the game's story can result in hundreds of save files that add up to a couple GBs, and slows down the game's Load page as it has to read each and every save file in the local folder to (very slowly) render the list to the player.

What many of us have done is simply delete unwanted files manually via WIndows Explorer (not in game), or move them elsewhere so the game can quickly load the list of most recent saves. That or disabling Cloud Steam midway results in local and remote going out of sync. And there appears to be no way - at least so far from my game play with [b]The WItcher EE[/b] via Steam to perform housekeeping or re-sync.

http://steamcommunity.com/app/20900/discussions/0/1457328392103556339/

I am stumped how to clean this up properly because I have read of horror stories of players getting ALL SAVE FILES CORRUPTED when quota limit is exceeded.

**Comment by GMMan** [comment]
Autocloud is a part of Steam Cloud, but does not expose its files through the API. It is an automated synchronization that runs before the game launches and after it exits, on a set of predefined paths. The only way to really affect the files that get synchronized is to modify the files while the game is running.

All you have to do is launch the game, manage the files being watched as declared in the sync paths, and exit the game to sync. Don't touch anything in Steam's userdata folder, because that's bound to cause sync problems.

**Comment by icelava** [comment]
Well the files are definitely out of sync. Before I disabled Cloud Steam, I have been deleting redundant save files in Windows Explorer because doing so in game has been painfully slow simply to read/load all of them onto the screen + terrible unscalable scroll bar makes for tedious file selection. So there is this whole stack of earlier save files in remote storage I have no convenient way of removing since the local side no longer have them.

The remotecache.vdf will show their entries no matter what I save or delete locally with Steam Cloud re-enabled. I can edit or delete remotecache.vdf - the only file in the 20900 folder and it will be restored according to the state in remote storage. No game-stopping error there.

Perhaps since I am _so close_ to exceeding that 947MB limit I might want to just "trigger" that and see if my whole game truly implodes (as described by others) because of it. I am making local backups of save files of course in preparation of this.

**Comment by icelava** [comment]
So I've just played **The Witcher** (Enhanced Edition) story to completion. Looks like the current version is more robust and does not shoot itself in the foot when the save files are messily out of sync and the quota exceeds. The game, or Steam Cloud, somehow manages to keep running without breaking; I see the remotecache.vdf updating itself with new save files as I progress incrementally through the chapters.

I saw on occasion though that the remote storage got reduced to half (down to like 400-500MB) but the remotecache,vdf still listed the files of the very first saves from the beginning prologue chapter and so forth. The local side does not have those files any more (deleted manually). Without having an in-depth of how to game is trying to sync local saves to remote storage, I guess this is a problem I can leave alone since at least it's not corrupting the integrity of the save files (at least from the ones I tired).

With that I don't think there is further need for me to utilise this tool. Thanks.
